sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/CreateHiveTableAsSelectCommand.scala

```diff
@@ -45,6 +46,11 @@ case class CreateHiveTableAsSelectCommand(
 
   override def run(sparkSession: SparkSession, child: SparkPlan): Seq[Row] = {
```

Comment:
Some more thoughts:

`CreateHiveTableAsSelectCommand` just runs another command, so we will not get any metric for this plan node. It's OK if we use the hive writer, as we indeed can't get any metrics(the writing is done by hive). However, if we can convert and use Spark's native writer, we do have metrics. I think a better fix is to replace Hive CTAS with data source CTAS during optimization.

Reply:
I think the table metadata created by data source CTAS and Hive CTAS are different?

Reply:
then how about we create a special Hive CTAS command that follows data source CTAS command but creates Hive table?

Reply:
I also thought about it. But then we will have two Hive CTAS commands. Is it good for you?

Reply:
I'm OK with that, since we do have 2 different ways to do Hive CTAS.

Reply:
I created a Hive CTAS with data source command.